tests/sources/test_notion.py

```diff
@@ -548,3 +556,126 @@ async def test_fetch_children_recursively():
                     block["paragraph"]["text"][0]["plain_text"]
                     == "This is a test paragraph block."
                 )
+
+
+@pytest.mark.parametrize(
+    "filtering",
+    [
+        Filter(
+            {
+                ADVANCED_SNIPPET: {
+                    "value": {
+                        "database_query_filters": [
+                            {
+                                "filter": {
+                                    "property": "Task completed",
+                                    "title": {"equals": "true"},
+                                },
+                                "database_id": "database_id",
+                            }
+                        ],
+                        "searches": [
+                            {
+                                "filter": {"value": "database"},
+                                "query": "efghsd",
+                            }
+                        ],
+                    }
+                }
+            }
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_get_docs_with_advanced_rules(filtering):
+    async with create_source(
+        NotionDataSource,
+        notion_secret_key="secert_key",
+    ) as source:
+        documents = []
+        with patch.object(
+            NotionClient, "query_database", return_value=AsyncIterator([DATABASE])
+        ), patch.object(
+            NotionDataSource,
+            "retrieve_and_process_blocks",
+            return_value=AsyncIterator([(BLOCK, None)]),
+        ):
+            expected_responses_ids = [DATABASE.get("id"), BLOCK.get("id")]
+            async for docs, _ in source.get_docs(filtering):
+                documents.append(docs.get("id"))
+            assert documents == expected_responses_ids
+
+
+@pytest.mark.parametrize(
+    "advanced_rules, expected_validation_result",
+    [
+        (
+            {
+                "searches": [
+                    {
+                        "filter": {"value": "database"},
+                        "query": "Demo",
+                    }
+                ]
+            },
+            SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
```

Comment:
Can you add comments to each of these to explain why each is False? This will help us maintain/troubleshoot down the road.